sonar-batch/src/main/java/org/sonar/batch/test/TestableBuilder.java

```diff
@@ -17,22 +17,31 @@
  * along with this program; if not, write to the Free Software Foundation,
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
-package org.sonar.core.test;
+package org.sonar.batch.test;
 
+import javax.annotation.CheckForNull;
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.batch.fs.InputFile.Type;
+import org.sonar.api.batch.fs.internal.DefaultInputFile;
 import org.sonar.api.test.MutableTestable;
-import org.sonar.core.component.GraphPerspectiveLoader;
-import org.sonar.core.graph.BeanVertex;
+import org.sonar.batch.deprecated.perspectives.PerspectiveBuilder;
+import org.sonar.batch.index.BatchComponent;
 
-public class TestablePerspectiveLoader extends GraphPerspectiveLoader<MutableTestable> {
+public class TestableBuilder extends PerspectiveBuilder<MutableTestable> {
 
-  static final String PERSPECTIVE_KEY = "testable";
-
-  public TestablePerspectiveLoader() {
-    super(PERSPECTIVE_KEY, MutableTestable.class);
+  public TestableBuilder() {
+    super(MutableTestable.class);
   }
 
+  @CheckForNull
   @Override
-  protected Class<? extends BeanVertex> getBeanClass() {
-    return DefaultTestable.class;
+  public MutableTestable loadPerspective(Class<MutableTestable> perspectiveClass, BatchComponent component) {
+    if (component.isFile()) {
+      InputFile inputFile = (InputFile) component.inputPath();
+      if (inputFile.type() == Type.MAIN) {
```

Comment:
![New](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/new.png)![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png)Possible null pointer dereference in org.sonar.batch.test.TestableBuilder.loadPerspective(Class, BatchComponent) due to return value of called method[![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=findbugs%3ANP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE)